dashboard/app/models/user.rb

```diff
@@ -1646,6 +1646,30 @@ def associate_with_potential_pd_enrollments
     end
   end
 
+  # When creating an account, we want to look for any channels that got created
+  # for this user before they signed in, and if any of them are in our Applab HOC
+  # course, we will create a UserScript entry so that they get a course card
+  def generate_progress_from_storage_id(storage_id, script_name='applab-intro')
```

Comment:
The only reason I didn't make `applab-intro` hard coded is to make testing easier. Not sure if there's a better way to do this.

Reply:
That's probably the best way. Alternatively you could have a function that just returns this script name and stub it in the test.